qa/tasks/s3tests.py

```diff
@@ -377,18 +377,15 @@ def run_tests(ctx, config):
             args += ['REQUESTS_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt']
         else:
             args += ['REQUESTS_CA_BUNDLE=/etc/pki/tls/certs/ca-bundle.crt']
-        # civetweb > 1.8 && beast parsers are strict on rfc2616
         attrs = ["not fails_on_rgw", "not lifecycle_expiration", "not test_of_sts", "not webidentity_test"]
         if client_config.get('calling-format') != 'ordinary':
             attrs += ['not fails_with_subdomain']
         if not client_config.get('with-sse-s3'):
             attrs += ['not sse_s3']
        
-        if 'extra_attrs' in client_config:
-            attrs = client_config.get('extra_attrs') 
+        attrs += client_config.get('extra_attrs', [])
         args += ['tox', '--', '-v', '-m', ' and '.join(attrs)]
-        if 'extra_args' in client_config:
-            args.append(client_config['extra_args'])
+        args += client_config.get('extra_args', [])
```

Comment:
Just so I'm clear, why does this get of `extra_args` need to be done twice for `args`? In the dbstore case wouldn't `args` have the string "not fails_on_dbstore" twice in it's list at the end.

Reply:
one of the lines is extra_attrs which refers to the attrs we use for filtering with the pytest -m option

this second one is called extra_args, which can add arbitrary arguments to the pytest command line. the latter doesn't appear to be used by any suites atm